bbc/simorgh
# AV Embeds endpoint integration + styling

Part of https://jira.dev.bbc.co.uk/browse/WSTEAM1-1226 and https://jira.dev.bbc.co.uk/browse/WSTEAM1-1228

Overall changes
======
- Adds URL construction and fetch for AV embeds to the new API endpoint
- Moves AV route parsing into `constructPageFetchUrl` function
- Makes `service` an optional parameter in some functions as some AV embed routes do not have the service in the URL
- Adjusts styling to make AV player fullscreen
- Captions are not shown as they are not returned in the API response
- Removes `superResponsive` property from SMP config as we use CSS `aspect-ratio` to correctly height the player
- Adjusts the Next.js API route for fetching local data to allow multiple `optionalParams`, as `av-embeds` routes can have multiple optional parameters at the end
- Adds local fixture data for various `av-embeds` routes

Testing
======
1. Visit https://av-embeds-endpoint-integration--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=pages-avembeds-page--example&viewMode=story
2. Confirm player renders and takes up max width
3. Note the player won't play content as Storybook is not a valid domain for media playback
4. Run the new API locally with https://github.com/bbc/fabl-modules/pull/7514 and update your API environment variable to point to the local instance
5. Visit http://localhost.bbc.com:7081/serbian/cyr/av-embeds/srbija-68707945?renderer_env=live
6. Confirm the player renders and plays back correctly
7. Visit http://localhost.bbc.com:7081/news/av-embeds/67303123?renderer_env=live
8. Confirm the player renders and plays back correctly
9. Visit http://localhost.bbc.com:7081/russian/av-embeds/features-49881797/pid/p07q3wwl?renderer_env=live
10. Confirm the player renders and plays back correctly
11. Visit http://localhost:7081/russian/av-embeds/media-38886884
12. Confirm the player renders. Note this will not allow playback as this is using local fixture data
13. Visit http://localhost:7081/serbian/cyr/av-embeds/srbija-68707945
14. Confirm the player renders. Note this will not allow playback as this is using local fixture data

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
